spec/feature_set_spec.rb

```diff
@@ -1,87 +1,180 @@
+
+      it "returns the strategy result false" do
+        expect(feature_set.on?(:feature)).to be_false
       end
 
-      it 'should have the feature' do
-        feature_set.has?(:super_sweet_feature).should be_true
+      context "with feature set default true" do
+        before { feature_set.default = true }
+
+        it "returns the strategy result false" do
+          expect(feature_set.on?(:feature)).to be_false
+        end
+      end
+
+      context "with definition default true" do
+        before { feature_set << Flip::Definition.new(:feature, :default => true) }
+
+        it "returns the strategy result false" do
+          expect(feature_set.on?(:feature)).to be_false
+        end
       end
     end
 
-    context 'doesnt has key super_sweet_feature' do
-      it 'should have the feature' do
-        feature_set.has?(:super_sweet_feature).should be_false
+    context "with always-true strategy" do
+      before { feature_set.add_strategy TrueStrategy }
+
+      context "and no defaults" do
+        before { feature_set << Flip::Definition.new(:feature) }
+
+        it "returns the strategy result true" do
+          expect(feature_set.on?(:feature)).to be_true
+        end
+      end
+
+      context "with feature set default false" do
+        before { feature_set << Flip::Definition.new(:feature) }
+        before { feature_set.default = false }
+
+        it "returns the strategy result true" do
+          expect(feature_set.on?(:feature)).to be_true
+        end
+      end
+
+      context "with definition default false" do
+        before { feature_set << Flip::Definition.new(:feature, :default => false) }
+
+        it "returns the strategy result true" do
+          expect(feature_set.on?(:feature)).to be_true
+        end
       end
     end
-  end
 
+    context "with null strategy" do
+      before { feature_set.add_strategy NullStrategy }
+
+      context "and feature set default" do
+        let(:feature_set_default) { double }
+        before { feature_set.default = feature_set_default }
+
+        it "returns the feature set default" do
+          expect(feature_set.on?(:feature)).to be(feature_set_default)
+        end
+
+        context "when default is a proc" do
+          let(:proc_result) { double }
+          let(:feature_set_default) { proc { proc_result } }
+
+          it "returns the proc result" do
+            expect(feature_set.on?(:feature)).to be(proc_result)
+          end
+
+          context "with an arity of 1" do
+            let(:feature_set_default) { proc { |definition| definition } }
+
+            it "passes the definition to the proc" do
+              expect(feature_set.on?(:feature)).to be(definition)
+            end
+          end
+
+          context "with an arity of 2" do
+            let(:feature_set_default) { proc { |_, options| options } }
+
+            it "passes the options to the proc" do
+              expect(feature_set.on?(:feature, :a => :b)).to eq(:a => :b)
```

Comment:
Yep, fair. Just trying to illustrate that the definition and options are available inside the default procs, which is preserved behaviour from DeclarationStrategy. Returning them up probably isn't super clear though, so I've changed it to return true based on the definition or options instead. Better?